lib/pbench/server/api/resources/query_apis/datasets_publish.py

```diff
@@ -38,23 +38,26 @@ def __init__(self, config: PbenchServerConfig, logger: Logger):
                 authorization=API_AUTHORIZATION.DATASET,
             ),
             action="update",
+            require_stable=True,
+            require_map=True,
         )
 
-    def generate_actions(self, params: JSON, dataset: Dataset) -> Iterator[dict]:
+    def generate_actions(
+        self, params: JSON, dataset: Dataset, map: dict[str, list[str]]
+    ) -> Iterator[dict]:
```

Comment:
Should the type hint for `params` be `ApiParams`, so that it matches the base class method (and the other overrider)?

Reply:
It should; got missed before, got missed again.